### PR TITLE
Add documentation for running unit tests under Valgrind

### LINKED ISSUE #8864: Easily enable valgrind in tests
One of the most useful features about the "memdebug" stuff is that it is easy to run the tests suite and find memory errors.  If we could easily run valgrind ```make test xxx=valgrind``` or a configure flag or such, it would reduce the need to build with that not-loved internal facility. :)  The hard part will be filtering the output to remove noise I think.

### PR DESCRIPTION
**Pull request by tmshort**
Fixes #8864; some unit tests do fail.

##### Checklist

- [X] documentation is added or updated

### CODE REVIEW

File: NOTES.VALGRIND [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Test behavior can be modified by adjusting envionment variables.
+
+EXE_SHELL
+
+This variable is used to specify the shell used to execute OpenSSL test
+programs. The default program (util/shlib_wrap.sh) initializes the
+enviroment to allow programs to find shared libraries. The variable can
+be modified to specify a different executable environment.
+
+   EXE_SHELL="$(pwd)/util/shlib_wrap.sh valgrind --error-exitcode=1 --leak-check=full -q"
```
Comment (richsalz) [inline NOTES.VALGRIND]: I suggest ``` `/bin/pwd` ``` instead of ```$(pwd)```

Comment (tmshort) [reply NOTES.VALGRIND]: `$PWD` also seems to work, but it depends on one's shell...

**Comment by tmshort** [comment]
How many ways can I spell environment? 3!

**Comment by levitte** [comment]
Merged.

e616c11e170ef524b12e218537f4bf290057f8b7 Add documentation for running unit tests under Valgrind

**Pull request closed without merging**